code/_onclick/click.dm

```diff
@@ -131,10 +132,12 @@
 	if(A in DirectAccess())
 		if(W)
 			W.melee_attack_chain(src, A, params)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)
 		else
 			if(ismob(A))
 				changeNext_move(CLICK_CD_MELEE)
 			UnarmedAttack(A)
+			SEND_SIGNAL(src, COMSIG_MOB_ATTACK, A, params)

```

Comment:
Put this signal in the UnarmedAttack proc and have a separate signal for unarmed

Reply:
will do